src/build/emitter.ts

```diff
@@ -1705,6 +1712,83 @@ export function emitWebIdl(
     }
   }
 
+  function emitAsyncIterator(i: Browser.Interface) {
+    function getAsyncIteratorSubtypes() {
+      if (i.iterator && i.iterator.kind === "iterable" && i.iterator.async) {
+        if (i.iterator.type.length === 1) {
+          return [convertDomTypeToTsType(i.iterator.type[0])];
+        }
+        return i.iterator.type.map(convertDomTypeToTsType);
+      }
+    }
+
+    function emitAsyncIterableDeclarationMethods(
+      i: Browser.Interface,
+      subtypes: string[],
+      paramsString: string
+    ) {
+      let methods;
+      if (subtypes.length === 1) {
+        // https://webidl.spec.whatwg.org/#value-asynchronously-iterable-declaration
+        const [valueType] = subtypes;
+        methods = [
+          {
+            name: "values",
+            definition: `AsyncIterableIterator<${valueType}>`,
+          },
+        ];
+      } else {
+        // https://webidl.spec.whatwg.org/#pair-asynchronously-iterable-declaration
+        const [keyType, valueType] = subtypes;
+        methods = [
+          {
+            name: "entries",
+            definition: `AsyncIterableIterator<[${keyType}, ${valueType}]>`,
+          },
+          {
+            name: "keys",
+            definition: `AsyncIterableIterator<${keyType}>`,
+          },
+          {
+            name: "values",
+            definition: `AsyncIterableIterator<${valueType}>`,
+          },
+        ];
+      }
+
+      const comments = i.iterator!.comments?.comment;
+
+      methods.forEach((m) => {
+        emitComments({ comment: comments?.[m.name] }, printer.printLine);
+        printer.printLine(`${m.name}(${paramsString}): ${m.definition};`);
+      });
+    }
+
+    const subtypes = getAsyncIteratorSubtypes();
+    if (subtypes) {
```

Comment:
Can this return early instead?

Reply:
I tried to be consistent with [`emitIterator()`](https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/fce78f39b24bd096483b96c446e6fc26d65ca086/src/build/emitter.ts#L1681-L1705), which doesn't use an early return either... 🤷‍♂️

Reply:
I think it's equally readable? It's like "hey, no subtypes and no method with sequences? We are done here then"